src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -440,6 +440,9 @@ private static List<Index> GetImpactedIndexesToWaitForToBecomeNonStale(DocumentD
             {
                 foreach (var index in database.IndexStore.GetIndexes())
                 {
+                    if (index.State is IndexState.Disabled or IndexState.Error)
```

Comment:
as discussed during the meeting, the Disabled indexes are obvious here, but Error ones (and Faulty) should not be filtered out.